docs/developer-docs/setup/pulling-canister-dependencies.md

```diff
@@ -0,0 +1,430 @@
+- Testing the accuracy of the integration and other canister code.
+- Performing tests without paying cycles.
+- Performing tests using non-production data and environments.
+- Performing tests with faster completion time when run locally. 
+
+To pull these canisters from the IC mainnet to be developed and tested using a local replica, the [`dfx deps`](/docs/references/cli-reference/dfx-deps.md) command and workflow can be used. 
+
+In this workflow, a **service provider** configures a canister to be `pullable`, then deploys the canister to the IC mainnet. A service provider can be DFINITY or it can be a community developer creating a public, third-party canister. 
+
+Then, a **service consumer** can pull the canister as a dependency from the canister directly from the mainnet and can deploy the dependencies on a local replica. 
+
+In this guide, we'll further describe this workflow and how to use it.
+
+## Determining if a canister should be `pullable`
+
+To pull a canister that's been configured as `pullable`, the service consumer only requires the canister ID. 
+
+Before pulling a canister, first the canister must be configured to be `pullable`. Additionally, we must propose the question of *should* the canister be `pullable`?
+
+#### `Pullable` examples:
+
+If a canister is providing a public service at a static canister ID, then it makes sense for the canister to be `pullable`.
+
+If a service canister also depends on other canisters, those dependencies should also be `pullable`. 
+
+#### Non-`pullable` examples:
+
+If the canister is meant for personal use and not intended for others to use, it does not make sense for the canister to be `pullable`. 
+
+If a canister wasm is published for other developers to use for deploying their own instance, it does not make sense for the canister to be `pullable` since the canister ID of the instance is not static; users can test integrations locally and deploy them directly. An example of this canister type is the asset canister generated by dfx. 
+
+## `Pullable` dfx.json example
+
+For a canister to be `pullable`, the `dfx.json` file must include a `pullable` definition. Below is an example:
+
+```json
+{
+  "canisters": {
+    "service": {
+      "type": "motoko",
+      "main": "src/pullable/main.mo",
+      "pullable": {
+        "dependencies": [],
+        "wasm_url": "https://github.com/lwshang/pullable/releases/latest/download/service.wasm",
+        "init_guide": "A natural number, e.g. 1"
+      }
+    }
+  }
+}
+```
+
+:::caution
+The wasm module of a `pullable` canister must be hosted via an URL so that service consumers can download it when pulling the dependency.
```

Comment:
I wonder if we should also add a footnote indicating that a future evolution of this feature will likely leverage direct wasm downloads from the replica.